actionview/test/template/url_helper_test.rb

```diff
@@ -15,6 +15,10 @@ class UrlHelperTest < ActiveSupport::TestCase
     get "/other" => "foo#other"
     get "/article/:id" => "foo#article", :as => :article
     get "/category/:category" => "foo#category"
+
+    scope :engine do
+      get "/" => "foo#bar"
```

Comment:
basically, with scope in routes, we will duplicate situation
```
# routes.rb
  mount MyEngine::Engine => '/my_engine'
```
Problem is that [here](https://github.com/rails/rails/blob/a72041597e7980a14926eafc09fbcbb9f8b1fd2c/actionview/lib/action_view/helpers/url_helper.rb#L560) we will get `url_string == '/engine'` and will try to compare it to `request_uri  == '/engine/'`.
I jump around stack, but could not figure out where exactly we get this difference. 
This is why I make fix in easy way.